ckanext/recombinant/views.py

```diff
@@ -91,14 +99,21 @@ def upload(id: str) -> Response:
         return h.redirect_to('recombinant.preview_table',
                              resource_name=resource_name,
                              owner_org=org['name'])
+    except KeyError as e:
```

Comment:
this is a really broad try:except: covering lots of code that could result in hiding an unrelated bug. Would be better to catch the KeyError inside _process_upload_file just on the lines where it occurs and raise a different exception that we can catch here.

Reply:
ahhh okay yeah makes sense.